runtime/drivers/duckdb/connectors.go

```diff
@@ -118,6 +121,86 @@ func (c *connection) ingest(ctx context.Context, env *connectors.Env, source *co
 	return summary, nil
 }
 
+func (c *connection) ingestMotherduckData(ctx context.Context, source *connectors.Source, env *connectors.Env) (*drivers.IngestionSummary, error) {
+	err := c.WithConnection(ctx, 1, func(ctx, ensuredCtx context.Context) error {
+		res, err := c.Execute(ctx, &drivers.Statement{Query: "SELECT current_database();"})
+		if err != nil {
+			return c.checkErr(err)
+		}
+		defer res.Close()
+
+		res.Next()
+		var localDB string
+		if err := res.Scan(&localDB); err != nil {
+			return c.checkErr(err)
+		}
+
+		// get token
+		token := env.Variables["TOKEN"]
+		if token == "" && env.AllowHostAccess {
+			token = os.Getenv("motherduck_token")
+		}
+		// load motherduck extension; connect to motherduck service
+		err = c.Exec(ctx, &drivers.Statement{Query: "LOAD 'motherduck';", Priority: 1})
+		if err != nil {
+			return fmt.Errorf("failed to load motherduck extension %w", err)
+		}
+
+		err = c.Exec(ctx, &drivers.Statement{Query: fmt.Sprintf("PRAGMA MD_CONNECT('token=%s');", token), Priority: 1})
+		if err != nil {
+			return fmt.Errorf("failed to load motherduck extension %w", err)
+		}
+
+		// get list of all motherduck databases
+		res, err = c.Execute(ctx, &drivers.Statement{Query: "SELECT name FROM md_databases();", Priority: 1})
+		if err != nil {
+			return c.checkErr(err)
+		}
+		defer res.Close()
+
+		names := make([]string, 0)
+		for res.Next() {
+			var name string
+			if res.Scan(&name) != nil {
+				return c.checkErr(err)
+			}
+
+			names = append(names, name)
+		}
+		if len(names) == 1 { // single motherduck db, use db to allow user to run query without specifying db name
```

Comment:
Should we consider allowing people to pass `use: name` in the YAML for cases where there are multiple DBs?

Reply:
Yeah can be done. Should we name the field `db` instead. 
If there are multiple db we can parse error to give nice instructions to use this field for table not found kind of errors.

Reply:
Although duckdb's error is self explanatory as well. 
```
/sources/md.yaml: Catalog Error: Table with name bids_data does not exist!
Did you mean "my_db.bids_data"?
```

Reply:
Both `use` or `db` or `database` seem fine to me, you decide

DuckDB's error seems adequate to me